src/modules/utils/panel/panels/ST7565.cpp

```diff
@@ -329,30 +341,41 @@ void ST7565::setContrast(uint8_t c)
     send_commands(contrast_seq, sizeof(contrast_seq));
 }
 
+/**
+*@brief Draws a character to the screen buffer
+*@param x   X coordinate
+*@param y   Y coordinate
+*@param c   Character to print
+*@param color 0: Turn pixels on (OR logic), 1: Turn pixels off (AND logic)
+*/
 int ST7565::drawChar(int x, int y, unsigned char c, int color)
```

Comment:
This no longer works when we do this without clearing the screen..
https://github.com/Smoothieware/Smoothieware/blob/edge/src/modules/utils/panel/screens/ModifyValuesScreen.cpp#L95
It used to clear the area where it was going to update the new value, this no longer works.

Reply:
Original smoothie code doesn't clear the screen only when values are changing and it was working correct only because drawChar() function was creating characters with background. Without this "background" we would see that the screen is not cleared but just adding character on top of character. Without @unlimitedbacon ["Fixed drawChar to not blank nearby pixels"](1a1049bbdb1449d4f032544fb1b6620f6c5586c9) commit the new gui WatchScreen is all overlapped with the "background" of displayed characters. 

I've added this 3 lines to [ModifyValuesScreen.cpp](https://github.com/mwr666/Smoothieware/blob/0d1c19ebdb910c329871fe7dd6b4baa09e311c35/src/modules/utils/panel/screens/ModifyValuesScreen.cpp#L95) and same 3 lines to [ControlScreen.cpp](https://github.com/mwr666/Smoothieware/blob/0d1c19ebdb910c329871fe7dd6b4baa09e311c35/src/modules/utils/panel/screens/ControlScreen.cpp#L60)

Reply:
I'd prefer this to be fixed without having to change ModifyValuesScreen.cpp or ControlScreen.cpp.
The changes to the ST7565 need to be backward compatible with the old behavior, otherwise there are a lot more places things need to be fixed. If it can't be made backwards compatible I will need to reconsider if this can be merged. 
Maybe a parameter can be passed to say if the write is opaque or transparent.

Reply:
I think the new function by default should work as original drawChar() code and with some additional flag turn transparent background on?

Reply:
yes agreed